src/logic/claims/data.ts

```diff
@@ -329,3 +361,64 @@ function getDisputeStatus(period: DisputePeriod): DisputeStatus {
       return 'Solved';
   }
 }
+
+async function loadClaimPayoutData(contract: ClaimsManager, params: { userAccount?: string; claimIds?: string[] }) {
+  const { userAccount = null, claimIds = null } = params;
+  const [acceptedEvents, settlementEvents, resolvedClaimEvents, resolvedSettlementEvents] = await Promise.all([
+    contract.queryFilter(
+      contract.filters.AcceptedClaim(
+        // @ts-expect-error
+        claimIds,
+        userAccount
+      )
+    ),
+    contract.queryFilter(
+      contract.filters.AcceptedSettlement(
+        // @ts-expect-error
+        claimIds,
+        userAccount
+      )
+    ),
+    contract.queryFilter(
+      contract.filters.ResolvedDisputeByAcceptingClaim(
+        // @ts-expect-error
+        claimIds,
+        userAccount
+      )
+    ),
+    contract.queryFilter(
+      contract.filters.ResolvedDisputeByAcceptingSettlement(
+        // @ts-expect-error
+        claimIds,
+        userAccount
+      )
+    ),
+  ]);
+
+  const events = sortEvents([
+    ...acceptedEvents,
+    ...settlementEvents,
+    ...resolvedClaimEvents,
+    ...resolvedSettlementEvents,
+  ] as const);
+
+  const byId = events.reduce((acc, ev) => {
+    if ('clippedAmountInUsd' in ev.args) {
```

Comment:
You probably want to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty

Reply:
Using `in` is a common way to narrow down types in Typescript: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing

Reply:
Yeah, but I dislike the fact that `in` traverses the full prototype chain and in some cases results in wrong logic.

Reply:
I think this is a matter of preference 🙂. I highly doubt we are at risk of any wrong logic here ito inherited properties.
Also, using `hasOwnProperty()` doesn't give us the type narrowing we want here:
<img width="673" alt="Screenshot 2022-09-29 at 12 48 23" src="https://user-images.githubusercontent.com/747979/193012250-7594a799-318a-47bf-9f2a-536165f4cb38.png">

Lastly, there is also the Eslint rule against using `hasOwnProperty()` directly: https://eslint.org/docs/latest/rules/no-prototype-builtins

Reply:
Yeah, feel free to leave it as is. Especially the narrowing is a big plus.